spec.md

```diff
@@ -1103,6 +1313,46 @@ If a `CreateVolume` operation times out, leaving the CO without an ID with which
 
 It is NOT REQUIRED for a controller plugin to implement the `LIST_VOLUMES` capability if it supports the `CREATE_DELETE_VOLUME` capability: the onus is upon the CO to take into consideration the full range of plugin capabilities before deciding how to proceed in the above scenario.
 
+##### `CreateSnapshot`, `DeleteSnapshot`, `ListSnapshots`
+
+The plugin-generated `snapshot_id` is a REQUIRED field for the `DeleteSnapshot` RPC, as opposed to the CO-generated snapshot `name` that is REQUIRED for the `CreateSnapshot` RPC: these fields MAY NOT contain the same value.
```

Comment:
based on discussion above. may be we should make it clear

```
CO-generated snapshot `name` specified in CreateSnapshot cannot be looked up by `ListSnapshots`. It is a write only field and is not used anywhere else.
```

Reply:
Sure.  I added some text to clarify this.  Thanks!

Reply:
this is a bit misleading. the text "... MAY NOT contain the same value." means this:

(a) the returned `snapshot_id` might be the same as the user-generated snapshot `name`
(b) .. or it could be completely different

to say that the ListSnapshots can't be filtered by the user-generated `name` only applies for snapshots that generate their own `snapshot_id` that's guaranteed to never match user-generated `name`. The sentence that was added below (`SHALL NOT be used...`) to resolve complaints is semantically inconsistent - please rewrite, or else remove it.